codingChallenge/imports/components/map/map.js

```diff
@@ -0,0 +1,134 @@
+                        zoom: 11,
+                        mapTypeId: google.maps.MapTypeId.ROADMAP,
+                    };
+
+                    function initMap() {
+                        if (map === void 0) {
+                            map = new google.maps.Map(document.getElementById('mviewer'), mapOptions);
+                        }
+                    }
+
+                    // Marker's Seting function  
+                    function setMarker(map, position, title, desc) {
+                        var marker;
+                        var markerOptions = {
+                            position: position,
+                            map: map,
+                            title: title,
+                            icon: 'https://maps.google.com/mapfiles/ms/icons/orange-dot.png'
+                        };
+
+                        marker = new google.maps.Marker(markerOptions);
+                        markers.push(marker); // add marker to array
+
+                        google.maps.event.addListener(marker, 'click', function() {
+                            
+                            // close window if not undefined
+                            if (infoWindow !== void 0) {
+                                infoWindow.close();
+                            }
+                            // create new info window
+                            var infoWindowOptions = {
+                                content: `<h2>${title}</h2><p>${desc}</p>`
+                            };
+                            infoWindow = new google.maps.InfoWindow(infoWindowOptions);
+                            infoWindow.open(map, marker);
+                        });
+                    }
+
+                    initMap();
+
+                    geocoder.geocode({ 'latLng': latlng },
+                        function(results, status) {
+                            if (status == google.maps.GeocoderStatus.OK) {
+                                if (results[0]) {
+                                    var add = results[0].formatted_address;
+                                    var value = add.split(",");
+                                    // Remove extra words in name
+                                    $scope.state = state = (value[value.length - 2]).replace(/governorate|city|state/gi,"").trim();
+                                    var eventsByLocation = `https://graph.facebook.com/search?q=${state}&type=event&center=${lat},${lng}&distance=${distance}&access_token=${fbToken}`;
+
+                                    $http.get(eventsByLocation)
+                                        .then(function(response) {
+                                            var events = response.data.data;
+                                            events.forEach(function(event, index) {
+                                                // check if event has a specified location
+                                                if (event.place.location) {
```

Comment:
@AbdallahZac  Some events doesn't have location. Should protect your condition further.
`event.place && event.place.location`  should do.